barretenberg/cpp/src/barretenberg/crypto/merkle_tree/node_store/cached_content_addressed_tree_store.hpp

```diff
@@ -629,8 +644,42 @@ fr ContentAddressedCachedTreeStore<LeafValueType>::get_current_root(ReadTransact
 // It is assumed that when these operations are being executed that no other state accessing operations
 // are in progress, hence no data synchronisation is used.
 
+template <typename LeafValueType> void ContentAddressedCachedTreeStore<LeafValueType>::commit_genesis_state()
+{
+    // In this call, we will store any node/leaf data that has been created so far
+    bool dataPresent = false;
+    TreeMeta meta;
+    // We don't allow commits using images/forks
+    if (forkConstantData_.initialised_from_block_.has_value()) {
+        throw std::runtime_error("Committing a fork is forbidden");
+    }
+    get_meta(meta);
+
+    auto currentRootIter = nodes_.find(meta.root);
+    dataPresent = currentRootIter != nodes_.end();
+    {
+        WriteTransactionPtr tx = create_write_transaction();
+        try {
+            if (dataPresent) {
+                persist_leaf_indices(*tx);
+                persist_node(std::optional<fr>(meta.root), 0, *tx);
+            }
+
+            meta.committedSize = meta.size;
+            persist_meta(meta, *tx);
+            tx->commit();
+        } catch (std::exception& e) {
+            tx->try_abort();
+            throw std::runtime_error(
+                format("Unable to commit genesis data to tree: ", forkConstantData_.name_, " Error: ", e.what()));
+        }
+    }
+    // rolling back destroys all cache stores and also refreshes the cached meta_ from persisted state
+    rollback();
+}
+
 template <typename LeafValueType>
-void ContentAddressedCachedTreeStore<LeafValueType>::commit(TreeMeta& finalMeta, TreeDBStats& dbStats, bool asBlock)
+void ContentAddressedCachedTreeStore<LeafValueType>::commit_block(TreeMeta& finalMeta, TreeDBStats& dbStats)
```

Comment:
Got rid of the `asBlock` arg. This was always tru except wjhen committing genesis state, which is now a dedicated method.